src/onyx/peer/task_lifecycle.clj

```diff
@@ -461,6 +479,26 @@
   (when (= (:onyx/type entry) :function)
     (operation/kw->fn (:onyx/fn entry))))
 
+(defn instantiate-plugin-instance [class-name pipeline-data]
+  (.newInstance (.getDeclaredConstructor ^Class (Class/forName class-name)
+                                         (into-array Class [clojure.lang.IPersistentMap]))
+                (into-array [pipeline-data])))
+
+(defn build-pipeline [task-map pipeline-data]
+  (let [kw (:onyx/ident task-map)]
+    (try 
+      (if (#{:input :output} (:onyx/type task-map))
+        (let [user-ns (namespace kw)
+              user-fn (name kw)]
+          (if (and user-ns user-fn)
+            (if-let [f (ns-resolve (symbol user-ns) (symbol user-fn))]
+              (f pipeline-data)    
+              (throw (Exception.)))
```

Comment:
True, though it is being caught and a new, better error is thrown. I'll improve it since we do return the inner exception in the ex-info now.